### DISCUSSION TITLE
action paramter for genericmiot (rice cooker profile)

### DISCUSSION THREAD

**Discussion by sayzard** [rytilahti/python-miio#1746] [Q&A]
Hi, 
I'm trying to turn on the heating mode of a Rice Cooker(chunmi.cooker.wy3).
The cancel action without a parameter works well.
However, If I specify a profile to start cooking,I  got the following error:

...
  File "/usr/local/lib/python3.8/dist-packages/construct/core.py", line 701, in _parse
    return self._decode(obj, context, path)
  File "/usr/local/lib/python3.8/dist-packages/miio/protocol.py", line 201, in _decode
    raise PayloadDecodeException(
miio.exceptions.PayloadDecodeException: Unable to parse message payload

my code:
...
profile = "0110050000000b24001e0023001900801200000002aa0000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000000000000000000280a003c000f91416e080000008278736e0
01000756e00690f0c75506e0669100082826e0869100069025004129d"
res=g_cooker.call_action_by(3,1,[profile])
....

The profile value was captured on my iPhone.

{"accessKey":"xxxx","params":{"did":"xxx","siid":3,"in":["0110050000000b24001e0023001900801200000002aa0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000280a003c000f91416e080000008278736e001000756e00690f0c75506e0669100082826e0869100069025004129d"],"aiid":1}}'

How should I specify the profile parameter in call_action?

Thanks in advance

**Comment by rytilahti** [comment]
I just tested the code you gave (against a simulator) and it's working as expected. 

As your stacktrace is cut short, I cannot see from where the error you are receiving is coming from. Maybe it is the response of the device that cannot be parsed? You could enable debug logging in your script to verify what is being sent & received:
```
import logging
logging.basicConfig(level=logging.DEBUG)

..

```

**Reply by sayzard** [reply]
I'm sorry for the lack of information in the previous post.

The response is strange,

----
DEBUG:miio.integrations.genericmiot.genericmiot:	SensorDescriptor(id='custom:taste', name='taste', type=<class 'int'>, property='custom_taste', unit=None)
DEBUG:miio.integrations.genericmiot.genericmiot:Created 0 settings
DEBUG:miio.miioprotocol:192.168.0.188:54321 >>: {'id': 2,
 'method': 'action',
 'params': {'aiid': 1,
            'did': 'call-3-1',
            'in': ['0110050000000b24001e0023001900801200000002aa0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000280a003c000f91416e080000008278736e001000756e00690f0c75506e0669100082826e0869100069025004129d'],
            'siid': 3}}
ERROR:miio.protocol:Unable to parse json 'b'{"id":2,"result":,"exe_time":0}'': Expecting value: line 1 column 18 (char 17)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/miio/protocol.py", line 194, in _decode
    return json.loads(decoded)

---

For reference,If I send it via micloud, I got the following response:
Request:
{'data': '{"accessKey": "xxxx", "params": {"did": "xxx", "siid": 3, "in": ["0110050000000b24001e0023001900801200000002aa0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000280a003c000f91416e080000008278736e001000756e00690f0c75506e0669100082826e0869100069025004129d"], "aiid": 1}}'}

Response:
{"code":0,"message":"ok","result":{"did":"xx","miid":0,"siid":3,"aiid":1,"code":0,"exe_time":80,"net_cost":119,"otlocalts":1677364539879792,"_oa_rpc_cost":202,"withLatency":0}}

**Reply by rytilahti** [reply]
Looks like this device has a buggy behavior when controlled locally, the response looks  incomplete and broken. Unfortunately, there is nothing you or this library can do in this situation :(
